loopy/codegen/result.py

```diff
@@ -61,6 +61,10 @@ class GeneratedProgram(Record):
         Once generated, this captures the AST of the operative function
         body (including declaration of necessary temporaries), but not
         the overall function definition.
+    .. attribute:: decl_ast
+        Once generated, this captures the AST of the operative function
+        definition but not the function body or declaration of necessary
+        temporaries.
```

Comment:
Adding it to the GeneratedProgram was the easiest way I saw to keep it around such that it was accessible from the codegen_result / device programs.  Another way would be fine.